### PR TITLE
all_first_color - add discussion of "gray areas", another possible solution & another challenge

### PR DESCRIPTION
**Pull request by tgaff**

### CODE REVIEW

**Comment by nathanallen** [comment]
It's a nice nugget, but easily missed. Perhaps we could use the commentary to intro `challenges/day2.md`?

I actually actively discourage them from considering edge-cases until they've solved the simplest cases first. Keeps them from overcomplicating their solutions.

I like the idea of having multiple accepted solutions. :+1:

File: challenges/solutions/all_first_color.js [hunk 1]
```diff
@@ -18,3 +18,34 @@ for(var j=0; j<5; j++) {
   draw();
   moveRight();
 }
+
+
+/* 
+Note that the above solution actually moves off the Kyrel board.  Maybe that's OK.
+
+Sometimes in software development the instructions you're given are not 100% clear.  There are frequently edge-cases, or other possibilities that hadn't been thought of yet.
```
Comment (nathanallen) [inline challenges/solutions/all_first_color.js:26]: Move to day2.md as problem-set intro.

File: challenges/solutions/all_first_color.js [hunk 2]
```diff
@@ -18,3 +18,34 @@ for(var j=0; j<5; j++) {
...
+
+for(var j=0; j<4; j++) {
+  moveRight();
+  draw();
+}
+
+*/
+
+/*
+Additional challenge:  What if the first square is empty?  Should we move that to the end as well and make the last square empty?
```
Comment (nathanallen) [inline challenges/solutions/all_first_color.js:49]: Not sure bonuses should go in the solution.

File: challenges/solutions/all_first_color.js (hunk 1 above)
Comment (tgaff) [inline challenges/solutions/all_first_color.js:26]: You need a solution to discuss this.  It's not going to be clear on day 2 without that.

This is the stuff that I talk about with them (in their pairs) and then later work through in front of them on the whiteboard.

File: challenges/solutions/all_first_color.js (hunk 2 above)
Comment (tgaff) [inline challenges/solutions/all_first_color.js:49]: I use these to challenge pairs that are a little bit ahead, but I want them to think of these questions on their own first if possible.  If I put it in the challenges then they won't have that opportunity.  The best place for this is in the "lecture notes", but we don't save those.

**Comment by nathanallen** [comment]
Perhaps updating the slides, then, is the solution?

I'm of a mind to keep the Kyrel repo very neutral with regards to how it's taught.

**Pull request closed without merging**